### PR TITLE
Unify all types of sliders with nouislider

### LINKED ISSUE #2527: Rendered slider* plots have missing bk-* classes
May happen on more than just sliders.

```
s = Slider(start=0, end=92, value=1, step=1, title="Time", callback=callback)
show(vplot(s))
```

renders as:
screen shot 2015-07-08 at 9 33 59 am

but `script, div = components(s)` outputs code that renders as
screen shot 2015-07-08 at 9 36 36 am

### LINKED ISSUE #5502: Consider nouislider to replace all our slider widgets
Available at https://refreshless.com/nouislider/. It's small, feature full and under liberal license. Doesn't use jquery-ui, which I would like to get rid of as much as possible.

### PR DESCRIPTION
**Pull request by mattpap**
- [ ] restore title attribute

fixes #5502
fixes #2527

### CODE REVIEW

**Comment by mattpap** [comment]
This is "ready". There are many things that can be improved about how sliders work, but I will be doing this in more fine-grained PRs. There's a new example `models/sliders.py` showing how all this works (note this example also nicely shows how bad layout is with widgets).

**Comment by bryevdv** [comment]
Some docs examples are failing with
```
AttributeError: unexpected attribute 'title' to Slider, possible attributes are bar_color, callback, callback_policy, callback_throttle, css_classes, direction, disabled, end, format, height, js_callbacks, name, orientation, sizing_mode, start, step, tags, tooltips, value or width
```

**Comment by bryevdv** [comment]
Several slider examples tests are failing with the same error:

https://s3.amazonaws.com/bokeh-travis/194998414/examples.log

File: bokeh/models/widgets/sliders.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    step = Int(default=1, help="""
+    The step between consecutive values.
+    """)
+
+    format = Override(default="%d %b %G")
+
+class DateRangeSlider(AbstractSlider):
+    """ Slider-based date range selection widget. """
+
+    value = Tuple(Date, Date, help="""
```
Comment (bryevdv) [inline bokeh/models/widgets/sliders.py]: same comment as above

File: bokeh/models/widgets/styles.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+Transparent = Enum("transparent")
+
+BorderStyle = Enum("none", "hidden", "dotted", "dashed", "solid", "double", "groove", "ridge", "inset", "outset", default=None)
+BorderWidth = Or(Length, Enum("thin", "medium", "thick"), default=None)
+
+def Tupled_1_4(tp, default=None):
+    return Or(tp, Tuple(tp, tp), Tuple(tp, tp, tp), Tuple(tp, tp, tp, tp), default=default)
+
+def OneOrTwo(tp, default=None):
+    return Or(tp, Tuple(tp, tp), default=default)
```
Comment (bryevdv) [inline bokeh/models/widgets/styles.py]: Do all of the above need to be public? Some seem pretty obscure.

File: bokeh/models/widgets/styles.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    display = Enum("none", "inline", "block", "inline-block", default=None) # TODO
+    visibility = Enum("visible", "hidden", "collapse", default=None)
+
+    white_space = Enum("normal", "pre", "nowrap", "pre-wrap", "pre-line", default=None)
+
+    word_spacing = Or(Enum("normal"), Length, Percentage, default=None)
+    letter_spacing = Or(Enum("normal"), Length, default=None)
+
+    z_index = Or(Auto, Int, default=None)
```
Comment (bryevdv) [inline bokeh/models/widgets/styles.py]: At a minimum, `Styles` class needs top level docstring. Ideally, all of the properties would have help that would link the relevant page on http://www.w3schools.com/cssref/ or similar

File: bokehjs/src/coffee/models/widgets/date_range_slider.coffee [hunk 4]
```diff
@@ -1,64 +1,28 @@
...
-      arrows:       [ p.Bool,   true   ]
-      value_labels: [ p.String, "show" ]
-      wheel_mode:   [ p.Any            ]
-      ###
-      formatter
-      scales
-      ###
-    }
+  behaviour: 'drag'
+  connect: [false, true, false]
```
Comment (bryevdv) [inline bokehjs/src/coffee/models/widgets/date_range_slider.coffee]: What is this? A comment for other / future devs would be appreciated

**Comment by bryevdv** [comment]
@mattpap I have left review comments. If you would like help with the docs ones, let me know.

**Comment by bryevdv** [comment]
Ran `models/sliders.py`:

screen shot 2017-01-25 at 1 12 37 pm

I didn't realize what the grey swatch was as first, I think some labeling would greatly help the comprehensibility of example:
* R, G, B for each slider
* "RGB color" for the patch

Does this PR already support adding a scale? If so, the example should represent that. If not, an issue should be made. That's an important and valuable feature of nouislider

The styling for the sliders is *extremely* bulky. I would prefer to try and make at least a rough  improvement to it in the first merge, and further refinements later.

File: bokeh/models/widgets/sliders.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    """)
+
+    format = Override(default="0,0.00")
+
+class RangeSlider(AbstractSlider):
+    """ Range-slider based number range selection widget. """
+
+    value = Tuple(Float, Float, help="""
+    Initial or selected range.
+    """)
```
Comment (mattpap) [inline bokeh/models/widgets/sliders.py]: No, but this was always implemented like this. I considered changing the API, but not necessarily in this PR.

**Comment by mattpap** [comment]
This PR was merged into PR #6335.

**Pull request closed without merging**